test/widgets/profile_test.dart

```diff
@@ -44,6 +45,8 @@ Future<void> setupPage(WidgetTester tester, {
     GlobalStoreWidget(
       child: MaterialApp(
         navigatorObservers: navigatorObserver != null ? [navigatorObserver] : [],
+        localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+        supportedLocales: ZulipLocalizations.supportedLocales,
```

Comment:
Mildly annoying that we have to do this, but oh well.

It does at least look like this doesn't introduce nondeterminism to the tests. I tried running them with a Japanese ambient locale:
```
$ LC_ALL=ja_JP.utf8 flutter test test/widgets/profile_test.dart
```
and they passed even though I'd added a check for the English version of a translated string:
```diff
--- a/test/widgets/profile_test.dart
+++ b/test/widgets/profile_test.dart
@@ -84,6 +84,7 @@ void main() {
 
       check(because: 'find user avatar', find.byType(Avatar).evaluate()).length.equals(1);
       check(because: 'find user name', find.text('test user').evaluate()).isNotEmpty();
+      check(find.text('Member').evaluate()).isNotEmpty();
     });
 
     testWidgets('page builds; profile page renders with profileData', (WidgetTester tester) async {
```

Reply:
I did test various situations on a device (the phone with different supported locales, or using a primary locale not in the supported list) and it seemed to behave nicely (no crashing, reverts to English - although when I look through the locale selection code it seems to try to find adjacent languages or languages from the same country). I didn't think to see if the ambient locale affected the tests!

But, I'm not sure if flutter is picking up on that locale setting. If I do `LC_ALL=ja_JP.utf8 flutter run` it doesn't appear to have picked up the locale.

Reply:
> But, I'm not sure if flutter is picking up on that locale setting.

If `flutter test` doesn't pick up on that setting, then that's great — what I was going for here was just that the tests won't behave differently on different people's development machines on account of their language settings.


> If I do `LC_ALL=ja_JP.utf8 flutter run` it doesn't appear to have picked up the locale.

Yeah, that seems appropriate — the app should be using the locale configured on the target device, not on the development host. … I guess it's less clear what it should do in the desktop case, though, where those are the same, and that's likely the scenario you were trying.